### PR TITLE
Refactor RARE arithmetic rules to avoid mixed arithmetic

### PR DESCRIPTION
**Pull request by ajreynol**
Ensures that mixed arithmetic cannot be introduced by RARE rules.  This fixes some known cases where mixed arithmetic appears even with `--proof-elim-subtypes`

FYI @Lachnitt

### CODE REVIEW

File: src/theory/arith/rewrites [hunk 1]
```diff
@@ -4,13 +4,13 @@
 ; interchangable. This is because Int and Real are not comparable in our
 ; type system.
 
-(define-rule arith-plus-zero ((t ? :list) (s ? :list)) (+ t 0 s) (+ t s))
```
Comment (abdoo8080) [inline src/theory/arith/rewrites:7]: Is this rule not needed anymore?

File: src/theory/arith/rewrites [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
-
 (define-rule arith-mul-one ((t ? :list) (s ? :list)) (* t 1 s) (* t s))
 (define-rule arith-mul-zero ((t ? :list) (s ? :list)) (* t 0 s) 0)
 
-(define-cond-rule arith-div-total ((t ?) (s ?)) (not (= s 0)) (/ t s) (/_total t s))
-(define-rule arith-div-total-zero ((x Real)) (/_total x 0/1) 0/1)
+(define-cond-rule arith-div-total-real ((t ?) (s Real)) (not (= s 0/1)) (/ t s) (/_total t s))
+(define-cond-rule arith-div-total-int ((t ?) (s Int)) (not (= s 0)) (/ t s) (/_total t s))
+(define-rule arith-div-total-zero-real ((t ?)) (/_total t 0/1) 0/1)
+(define-rule arith-div-total-zero-int ((t ?)) (/_total t 0) 0/1)
```
Comment (abdoo8080) [inline src/theory/arith/rewrites:13]: These rules look weird now with the type of one argument specified, but not the other. Is this done to keep supporting mixed arithmetic?

File: src/theory/arith/rewrites (hunk 1 above)
Comment (ajreynol) [reply src/theory/arith/rewrites:7]: Yes, it was created prior to ARITH_POLY_NORM.  There is no need to duplicate it for Int, Real.

File: src/theory/arith/rewrites (hunk 2 above)
Comment (ajreynol) [reply src/theory/arith/rewrites:13]: Yes, for now. Division can take either `Int` or `Real` in both arugments.

I am planning on eliminating mixed arithmetic from division at some point, at which case these rules can be simplified.

**Review by abdoo8080 — APPROVED** [review]

**Pull request merged**